AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -608,20 +669,22 @@ am_Error_e CAmSocketHandler::removeTimer(const sh_timerHandle_t handle)
     if (it == mListTimer.end())
         return (E_NON_EXISTENT);
 
-    close(it->fd);
+    mListRemovedTimers.push_back(*it);
     mListTimer.erase(it);
     return removeFDPoll(handle);
 #else
     stopTimer(handle);
     std::list<sh_timer_s>::iterator it(mListTimer.begin());
-    for (; it != mListTimer.end(); ++it)
+    while (it != mListTimer.end())
```

Comment:
I have to agree.  (Sorry for all the complains - we're just sharing ideas and experiences)  

Prefer **for** instead of **while**, because it feels more standard for iterating over the whole container (Yes, OK, there is a break condition, but notice the condition used in the loop itself is to continue until you reach the end!  This is typical of for-loops).

If you did something like 
```
while (it != end && it->handle != handle) 
{} 
```
then, maybe, **while** would make sense.   But why change?

Actually overall, the most important thing is consistency, and making this change only here makes it very *inconsistent*.  Better to look over the whole code base at some point.  Right now there is everything from for(), while and for_each with lambdas... used for the same type of loop.